tests/foreman/api/test_repository.py

```diff
@@ -1659,6 +1668,38 @@ def test_positive_synchronize(self, repo):
         repo.sync()
         assert repo.read().content_counts['docker_manifest'] >= 1
 
+    @pytest.mark.tier2
+    @pytest.mark.parametrize(
+        'repo_options',
+        **parametrized(
+            [
+                {
+                    'content_type': 'docker',
+                    'docker_upstream_name': CONTAINER_UPSTREAM_NAME,
+                    'name': gen_string('alphanumeric', 10),
+                    'url': CONTAINER_REGISTRY_HUB,
+                }
+            ]
+        ),
+        indirect=True,
+    )
+    def test_positive_delete_product_with_synced_repo(self, repo, repo_options_custom_product):
+        """Create and sync a Docker-type repository, delete the product.
+
+        :id: c3d33836-54df-484d-97e1-f9fc9e22d23c
+
+        :parametrized: yes
+
+        :expectedresults: A product with a synced Docker repository can be deleted.
+
+        :CaseImportance: High
+
+        :BZ: 1867287
+        """
+        repo.sync()
+        assert repo.read().content_counts['docker_manifest'] >= 1
+        assert repo.product.delete()
```

Comment:
Going to merge for the BZ coverage, but I think you should take a look at this module's test content with an eye on how efficient it is. This exact repository is created and synced for `test_positive_synchronize`, and factors like whether or not there is a custom product on the repo, what type of repo it is, etc make for great parameters.

Parametrization itself could be simplified, so that we're parametrizing key lists from dictionaries that hold core repository options. This provides better parameter contexts in the node ID (instead of the integers that the current parametrization style results in), and fewer test functions to maintain. A single test function can provide functional coverage for several actions without becoming overly complex or long.

While we'll repeat lots of API POST calls with slightly different data, we won't be doing it with slightly different test cases that have large chunks of repeated docblock and parametrization marks.

Reply:
@mshriver Thank you for the review and feedback. As the module is not mine (I was just doing this PR as the Jira card was still in my backlog) I just wanted to make a non-disruptive addition. As I am writing news test for 6.10 I can learn to make better use of parametrisation.

Reply:
ack, thanks @swadeley for adding the coverage!  One of those things that stuck out to me, and I figured it was better to call it when I see it :)